extensions/spark/setup/djl_spark/task/text/text2text_generator.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+#
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import pandas as pd
+from pyspark.sql.functions import pandas_udf
+from pyspark.sql.types import StringType
+from typing import Iterator
+from transformers import pipeline
+
+
+class Text2TextGenerator:
+
+    def __init__(self, input_col, output_col, engine, model_url=None, model_name=None):
+        """
+        Initializes the Text2TextGenerator.
+
+        :param input_col: The input column
+        :param output_col: The output column
+        :param engine: The engine. Currently only PyTorch is supported.
+        :param model_url: The model URL
+        :param model_name: The model name
+        """
+        self.input_col = input_col
+        self.output_col = output_col
+        self.engine = engine
+        self.model_url = model_url
+        self.model_name = model_name
+
+    def generate(self, dataset, **kwargs):
+        """
+        Performs text2text generation on the provided dataset.
+
+        :param dataset: input dataset
+        :return: output dataset
+        """
+        if not self.model_url and not self.model_name:
```

Comment:
Maybe check for not both

Reply:
Like: `if not any([self.model_url, self.model_name])`?

Reply:
Basically xor. Must have one or the other but not both. `bool(self.model_url) ^ bool(self.model_name)` should work.

Reply:
I was assuming both should be fine. Just model_url has higher priority than model_name.

Reply:
It is fine, but not great. For cases like this, I tend to ask whether a user would expect that this is the behavior of the API. If they did provide both, would they expect that priority? Instead, an error is clearer because having both properties when only a single one is used already means something is not right. The error will highlight that for the user and ensure they can clarify the situation rather then risk behaving in ways that they don't expect